### ISSUE TITLE
Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by kellyselden** [buxlabs/amd-to-es6#83]
using 0.12.0

source:

```js
define('ember-data/-debug/index', ['exports'], function (exports) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.instrument = instrument;
  function instrument(method) {
    return method();
  }

  /*
    Assert that `addedRecord` has a valid type so it can be added to the
    relationship of the `record`.
  
    The assert basically checks if the `addedRecord` can be added to the
    relationship (specified via `relationshipMeta`) of the `record`.
  
    This utility should only be used internally, as both record parameters must
    be an InternalModel and the `relationshipMeta` needs to be the meta
    information about the relationship, retrieved via
    `record.relationshipFor(key)`.
  
    @method assertPolymorphicType
    @param {InternalModel} internalModel
    @param {RelationshipMeta} relationshipMeta retrieved via
           `record.relationshipFor(key)`
    @param {InternalModel} addedRecord record which
           should be added/set for the relationship
  */
  var assertPolymorphicType = void 0;

  if (true) {
    var checkPolymorphic = function checkPolymorphic(modelClass, addedModelClass) {
      if (modelClass.__isMixin) {
        //TODO Need to do this in order to support mixins, should convert to public api
        //once it exists in Ember
        return modelClass.__mixin.detect(addedModelClass.PrototypeMixin);
      }
      if (Ember.MODEL_FACTORY_INJECTIONS) {
        modelClass = modelClass.superclass;
      }
      return modelClass.detect(addedModelClass);
    };

    exports.assertPolymorphicType = assertPolymorphicType = function assertPolymorphicType(parentInternalModel, relationshipMeta, addedInternalModel) {
      var addedModelName = addedInternalModel.modelName;
      var parentModelName = parentInternalModel.modelName;
      var key = relationshipMeta.key;
      var relationshipModelName = relationshipMeta.type;
      var relationshipClass = parentInternalModel.store.modelFor(relationshipModelName);
      var assertionMessage = 'You cannot add a record of modelClass \'' + addedModelName + '\' to the \'' + parentModelName + '.' + key + '\' relationship (only \'' + relationshipModelName + '\' allowed)';

      (true && !(checkPolymorphic(relationshipClass, addedInternalModel.modelClass)) && Ember.assert(assertionMessage, checkPolymorphic(relationshipClass, addedInternalModel.modelClass)));
    };
  }

  exports.assertPolymorphicType = assertPolymorphicType;
});
```

stack:

```
TypeError: Cannot read property 'name' of undefined
    at nodes.forEach.node (C:\code\ember-cli\node_modules\@buxlabs\amd-to-es6\src\class\Exporter.js:24:45)
    at Array.forEach (<anonymous>)
    at Exporter.getExports (C:\code\ember-cli\node_modules\@buxlabs\amd-to-es6\src\class\Exporter.js:19:11)
    at Exporter.harvest (C:\code\ember-cli\node_modules\@buxlabs\amd-to-es6\src\class\Exporter.js:13:17)
    at Module.convert (C:\code\ember-cli\node_modules\@buxlabs\amd-to-es6\src\class\Module.js:31:37)
    at module.exports (C:\code\ember-cli\node_modules\@buxlabs\amd-to-es6\src\converter.js:13:12)
    at Object.transform (C:\code\ember-cli\node_modules\rollup-plugin-amd\dist\rollup-plugin-amd.cjs.js:23:31)
    at C:\code\rollup\dist\rollup.js:18093:48
    at <anonymous>
```

**Comment by emilos** [comment]
Thanks! It's interesting how to handle things like:

```js
if (condition) {
  exports.a = 1
} else {
  exports.a = 2
}
```

To make it even more complex the example above does:

```js
var a = undefined
if (condition) {
  exports.a = a = 'foo'
}
exports.a = a
```

**Comment by emilos** [comment]
@kellyselden just to keep you updated on this one: the way I'll probably handle it is to:

1)

```js
var a = undefined
if (true) {
  exports.a = a = 'foo'
}
exports.a = a
```

2)

```js
var a = undefined
exports.a = a = 'foo'
exports.a = a
```

3)

```js
var a = undefined
a = 'foo'
exports.a = a
exports.a = a
```



4)

```js
var a = undefined
a = 'foo'
exports.a = a
```

It's not super hard, I just need some time to write the 2->3 transformation :) I should have some time next week

**Comment by kellyselden** [comment]
Thanks.

**Comment by emilos** [comment]
@kellyselden I've released a fix in `v0.12.3` - please let me know if it helps. Thanks again!
